### PR TITLE
Demonstrate creation or conversion of cluster to vnodes aligned on powers of 2

### PR DESCRIPTION
**Pull request by bhalevy**
The short series adds 2 unit tests that demonstrate the creation of a new cluster with tokens aligned on powers of 2
and then conversion of an existing cluster to such token map.

This can be used as part of a demonstration for migrating vnodes keyspace to tablets (https://scylladb.atlassian.net/browse/SCYLLADB-411)

Fixes https://scylladb.atlassian.net/browse/SCYLLADB-417

### CODE REVIEW

**Comment by bhalevy** [comment]
- renamed second patch

**Comment by bhalevy** [comment]
- fixed token verification
- refactored helper function to be used in both tests

**Comment by avikivity** [comment]
This should be part of a dedicated branch, not on master.

**Comment by bhalevy** [comment]
> This should be part of a dedicated branch, not on master.

Done.  I created `migration-from-vnodes-to-tablets` as the dedicated branch.

**Comment by avikivity** [comment]
> > This should be part of a dedicated branch, not on master.
> 
> Done. I created `migration-from-vnodes-to-tablets` as the dedicated branch.

Why is it on the main repo?

File: test/cluster/test_tablets.py [hunk 1]
```diff
@@ -1717,3 +1719,108 @@ async def test_table_creation_wakes_up_balancer(manager: ManagerClient):
         # up to stats refresh period, which is 60s. So use a small timeout.
         await manager.api.message_injection(server.ip_addr, 'wait-before-topology-coordinator-goes-to-sleep')
         await log.wait_for('wait-after-topology-coordinator-gets-event: wait', from_mark=mark, timeout=5)
+
+
+def calculate_powof2_tokens(num_nodes: int, tokens_per_node: int) -> dict[int, list[int]]:
+    exp = 1
```
Comment (ndragazis) [inline test/cluster/test_tablets.py]: Edge case: with `num_nodes=1` and `tokens_per_node=1` we get two tokens instead of one

File: test/cluster/test_tablets.py [hunk 2]
```diff
@@ -1717,3 +1719,108 @@ async def test_table_creation_wakes_up_balancer(manager: ManagerClient):
...
+        await asyncio.gather(*[manager.cql.run_async(stmt, [k, k]) for k in range(num_keys)])
+
+        calculated_new_tokens = calculate_powof2_tokens(num_nodes, tokens_per_node)
+
+        new_servers = []
+        for i, new_tokens in calculated_new_tokens.items():
+            new_server_id = num_nodes + i
+            logger.debug(f"Adding server {new_server_id} tokens={new_tokens}")
+            cmdline = [
+                f"--initial-token={','.join([str(t) for t in new_tokens])}",
```
Comment (ndragazis) [inline test/cluster/test_tablets.py:1781]: What if some new token is in use by a node? The new node won't bootstrap.

File: test/cluster/test_tablets.py [hunk 3]
```diff
@@ -1717,3 +1719,66 @@ async def test_table_creation_wakes_up_balancer(manager: ManagerClient):
...
+def calculate_powof2_tokens(num_nodes: int, tokens_per_node: int) -> dict[int, list[int]]:
+    exp = 1
+    while 2**exp < num_nodes * tokens_per_node:
+        exp += 1
+    new_tokens_combined = []
+    powof2 = 2**exp
+    for i in range(powof2):
+        pfx = (i - powof2 // 2) << (64 - exp)
+        if not new_tokens_combined:
+            pfx += 1  # avoid min token
```
Comment (tgrabiec) [inline test/cluster/test_tablets.py]: if we want ranges to be compatible with tablet ranges that scylla will generate for a given count, then this is the right formula:
```
def tokens(n):
     return [int(i * 2**64 // n - 2**63 - 1) for i in range(1, n+1)]
```

```
>>> tokens(4)
[-4611686018427387905, -1, 4611686018427387903, 9223372036854775807]
```

File: test/cluster/test_tablets.py (hunk 2 above)
Comment (bhalevy) [reply test/cluster/test_tablets.py:1781]: That should extremely rare, but indeed, it might happen. I think we can do another round of adjustments for the clashed tokens.
But let's keep that as a follow up.
Hopefully, we'll have functioning logic for arbitrary tablet boundaries by then, and it would solve this issue

File: test/cluster/test_tablets.py (hunk 3 above)
Comment (bhalevy) [reply test/cluster/test_tablets.py]: Done in https://github.com/bhalevy/scylla/pull/2 [88a9d7d](https://github.com/bhalevy/scylla/pull/2/commits/88a9d7de2fe4cc9d1394c82f6ada038ae48b12a2)

File: test/cluster/test_tablets.py (hunk 2 above)
Comment (bhalevy) [reply test/cluster/test_tablets.py:1781]: Added FIXME comment in https://github.com/bhalevy/scylla/pull/2 [f64c5ff](https://github.com/bhalevy/scylla/pull/2/commits/f64c5ffe4c8ad3466e366ac3eca09811723d8d1d)

File: test/cluster/test_tablets.py (hunk 1 above)
Comment (bhalevy) [reply test/cluster/test_tablets.py]: Fixed in https://github.com/bhalevy/scylla/pull/2 [f64c5ff](https://github.com/bhalevy/scylla/pull/2/commits/f64c5ffe4c8ad3466e366ac3eca09811723d8d1d)
